packages/cactus-api-client/src/main/typescript/socketio-api-client.ts

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2020-2021 Hyperledger Cactus Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * SocketIOApiClient.ts
+ */
+
+// TODO - Document it
+
+const defaultMaxCounterRequestID = 100;
+const defaultSyncFunctionTimeoutMillisecond = 5000;
+
+import { Logger, Checks } from "@hyperledger/cactus-common";
+import { LogLevelDesc, LoggerProvider } from "@hyperledger/cactus-common";
+import { ISocketApiClient } from "@hyperledger/cactus-core-api";
+
+import { Socket, SocketOptions, ManagerOptions, io } from "socket.io-client";
+import { readFileSync } from "fs";
```

Comment:
@outSH How will this work in a browser environment? Or is this class entirely only designed for the back-end?

Reply:
@petermetz Same as above, only backend support for now.

Reply:
@outSH Got it. When do plan on adding browser support in there?

Reply:
@petermetz It's hard to estimate at the moment, I create a new issue for this problem

Reply:
@outSH Cool, thank you. If you have the issue opened please send a link here as to that one as well (to establish the link/reference)

Reply:
@petermetz Done - https://github.com/hyperledger/cactus/issues/1675